sotodlib/site_pipeline/make_atomic_filterbin_map.py

```diff
@@ -0,0 +1,556 @@
+    parser.add_argument("--query",
+                        help='query, can be a file (list of obs_id) or selection string')
+    parser.add_argument("--odir",
+                        help='output directory')
+    parser.add_argument("--preprocess_config", type=str,
+                        help='file with the config file to run the preprocessing pipeline')
+    parser.add_argument('--update_delay', type=int,
+                        help="Number of days (unit is days) in the past to start observation list.")
+    parser.add_argument("--mode", type=str, )
+    parser.add_argument("--nproc", type=int, help='Number of procs in the multiprocessing pool')
+    parser.add_argument("--comps", type=str,)
+    parser.add_argument("--singlestream", action="store_true")
+    parser.add_argument("--only_hits", action="store_true") # this will work only when we don't request splits, since I want to avoid loading the signal
+    
+    # detector position splits (fixed in time)
+    parser.add_argument("--det_in_out", action="store_true")
+    parser.add_argument("--det_left_right", action="store_true")
+    parser.add_argument("--det_upper_lower", action="store_true")
+    
+    # time samples splits
+    parser.add_argument("--scan_left_right", action="store_true")
+    
+    parser.add_argument("--ntod", type=int, )
+    parser.add_argument("--tods", type=str, )
+    parser.add_argument("--nset", type=int, )
+    parser.add_argument("--wafer", type=str,
+                        help="Detector set to map with")
+    parser.add_argument("--freq", type=str,
+                        help="Frequency band to map with")
+    parser.add_argument("--max-dets", type=int, )
+    parser.add_argument("--fixed_ftime", type=int, )
+    parser.add_argument("--mindur", type=int, )
+    parser.add_argument("--site", type=str, )
+    parser.add_argument("--verbose", action="count", )
+    parser.add_argument("--quiet", action="count", )
+    parser.add_argument("--window", type=float, )
+    parser.add_argument("--dtype_tod", )
+    parser.add_argument("--dtype_map", )
+    parser.add_argument("--atomic_db",
+                        help='name of the atomic map database, will be saved where make_filterbin_map is being run')
+    parser.add_argument("--l2_data_path",
+                        help='Path to level-2 data')
+    return parser
+
+def _get_config(config_file):
+    return yaml.safe_load(open(config_file,'r'))
+
+def get_ra_ref(obs, site='so_sat3'):
+    # pass an AxisManager of the observation, and return two
+    # ra_ref @ dec=-40 deg.   
+    # 
+    #t = [obs.obs_info.start_time, obs.obs_info.start_time, obs.obs_info.stop_time, obs.obs_info.stop_time]
+    t_start = obs.obs_info.start_time
+    t_stop = obs.obs_info.stop_time
+    az = np.arange((obs.obs_info.az_center-0.5*obs.obs_info.az_throw)*putils.degree,
+                   (obs.obs_info.az_center+0.5*obs.obs_info.az_throw)*putils.degree, 0.5*putils.degree)
```

Comment:
Remember this plot that Atsuto made? We simply want to know where our footprint is relative to the horizontal direction. We just calculate the R.A. the footprint crosses at DEC=-40°. Of course if the footprint is in the North Galactic pole then probably it never crosses DEC = -40° so that why I had to get creative with the extrapolation stuff.
This is what you said about this plot

> Matthew Hasselfield
 It is definitely useful to have that kind of parameter!  Couple suggestions:
You could take the "RA of the boresight at dec=-40" as the parameter to group on.  It's not super trivial to compute but it is well defined.
A numbered sub_id (0, 1, 2, ...) suggests that we'll always observe in the same footprints.  I don't think that's the plan.  So maybe we can use some continuous instead of discrete scale.
It could be called ra_ref_min and ra_ref_max -- e.g. a scan might go from RA=12 deg to RA=72 deg (at dec=-40).

![image](https://github.com/user-attachments/assets/105c0a8a-be07-460a-b3eb-7e7e6e2a58f3)